pkg/operator/ceph/object/dependents.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2021 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package object
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	v1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/daemon/ceph/client"
+	"github.com/rook/rook/pkg/util/dependents"
+)
+
+// CephObjectStoreDependents returns the buckets which exist in the object store that should block
+// deletion.
+func CephObjectStoreDependents(
+	clusterdCtx *clusterd.Context,
+	clusterInfo *client.ClusterInfo,
+	store *v1.CephObjectStore,
+) (*dependents.DependentList, error) {
+	nsName := fmt.Sprintf("%s/%s", store.Namespace, store.Name)
+	baseErrMsg := fmt.Sprintf("failed to get dependents of CephObjectStore %q", nsName)
+
+	deps := dependents.NewDependentList()
+
+	objCtx, err := NewMultisiteContext(clusterdCtx, clusterInfo, store)
+	if err != nil {
+		return deps, errors.Wrapf(err, "%s. failed to get context", baseErrMsg)
+	}
+	opsCtx, err := NewMultisiteAdminOpsContext(objCtx, &store.Spec)
+	if err != nil {
+		return deps, errors.Wrapf(err, "%s. failed to get admin API context", baseErrMsg)
+	}
+
+	buckets, err := opsCtx.AdminOpsClient.ListBuckets(context.TODO())
+	if err != nil {
+		return deps, errors.Wrapf(err, "%s. failed to list buckets in CephObjectStore %q", baseErrMsg, nsName)
+	}
```

Comment:
A bit complex suggestion(feel free to ignore), if we can refer OB details to the existing bucket. Say if there are buckets a,b,c and a belongs to OB1, c belongs to OB2, b does not have any OB references. if it exists add the OBC into a new key-value pair for in the dependents list.
if OB1 exists
dep.Add("obc in the objectstore", OB1)
So it will help users/admin to delete them than information about the backend bucket.

Reply:
From the design doc, we chose to use only bucket names so that we don't have to use an external API for listing dependencies. I do agree that this would be an improvement in experience for the user, but I think this is a feature that we can add later if users are asking for it.